ostd/src/arch/x86/iommu/registers/mod.rs

```diff
@@ -204,6 +225,24 @@ impl IommuRegisters {
         while !self.read_global_status().contains(GlobalStatus::QIES) {}
     }
 
+    /// Enables DMA remapping based on the root table address register (`value`).
+    ///
+    /// # Safety
+    ///
+    /// The user must ensure:
+    /// 1. The correctness of the `value`.
+    /// 2. After enabling DMA remapping, there won't be soundness problems related to DMA.
```

Comment:
These two safety conditions are difficult to dispute in terms of their correctness. The safety conditions are provided so that readers can verify whether they are met on their own. Therefore, I think it is necessary to clarify them further.

```rust
/// # Safety
/// 
/// The user must ensure:
/// 1. The `value` is the physical address of a root table which lives long enough, 
/// i.e., the root table won't be dropped if the IOMMU is still enabled.
/// ......
```

The second point is hard to understand, so it would be best to make it clearer. If you think providing a complete description of the safety requirements is too lengthy, you can also include some references here.